action.yaml

```diff
@@ -54,6 +54,46 @@ runs:
 
         fi
 
+    - name: Test Report
+      uses: dorny/test-reporter@v1
+      id : test-result
+      if: ${{ env.cmd_type == 'UNIT_TEST' &&  env.test_report_path !='' && env.reporter !='' }}                   
+      with:
+        name: Unit Test                          
+        path: ${{ env.test_report_path}}              # '**/target/**/*.xml'   
+        reporter: ${{ env.reporter }}                 # Format of test results. 
+        only-summary: 'true' 
+        fail-on-error: ${{ env.fail-on-error || false }}     # Set action as failed if test report contains any failed test
+
+    - name: Copy test report to target folder
+      if: ${{ env.copy-to-target-path !='' }}
```

Comment:
Additionally to this condition , it has to have the same conditions as the `Test Report`. The copy test report step should only be executed if the unit tests were actually run.

Reply:
I have tried this condition` if: ${{ env.copy-to-target-path !=''  &&  env.cmd_type == 'UNIT_TEST' &&  env.test_report_path !='' && env.reporter !=''}}` but when we set the parameter as `fail-on-error==true`  then test-reports as well as report files are getting generated but the `Copy test report to target folder ` & `Fetch Unit Test Report Url `  steps are not getting executed as above step `Test Report` is failing.

So when the `Test Report` step fails due to `fail-on-error==true` or any other condition like wrong parameter values like ` test_report_path, reporter` are we supposed to copy reports to the target folder or not?

Reply:
If the tests reports are there, I don't see a problem on copying them even if the previous error failed but it continues due to `fail-on-error` being `true`. 

In [here](https://github.com/lumada-common-services/gh-composite-actions/pull/91/files#diff-fab4d7fb461bc6fbe9587f6c03fff98102b1c744145edcf2a993f2ff7cb05a0dR63), you're giving the example of `'**/target/**/*.xml'`, does this work on a `cp` statement? And what if the tests reports are not there, running the `cp` command anyway, what are the consequences?

Reply:
Yes `'**/target/**/*.xml'` works with the cp command here's the [Output](https://github.com/lumada-common-services/multi-module-tester/actions/runs/5784059929/job/15674799425#step:10:183). If `'test_report_path'` is empty then it will show the following [error](https://github.com/lumada-common-services/multi-module-tester/actions/runs/5796894265/job/15711370809#step:10:159) and for wrong `'test_report_path'`  it will show the following [error](https://github.com/lumada-common-services/multi-module-tester/actions/runs/5796604067/job/15711202908#step:10:194).

Reply:
> Yes '**/target/**/*.xml' works with the cp command here's the [Output](https://github.com/lumada-common-services/multi-module-tester/actions/runs/5784059929/job/15674799425#step:10:183).
But did you check if the files were actually copied to the target folder?